frappe/erpnext
# refactor: get/create customer for Sales Order

- Split one big method into three smaller ones
- Remove unused parameter `customer_group`. (_Customer Group_ should be present in **Lead** in order to get mapped to **Customer**.)
- Remove redundant handling of naming conflicts (the **Customer** controller already takes care of this and does a better job)
- Turn the rename notice into a toast, so that it seems less like an error message

How to test this PR:

- Create a **Lead** -> **Quotation** -> **Sales Order**. A **Customer** should get created automatically.
- Create another **Lead** with the same name and repeat above process. A **Customer** should get created automatically (named with suffix "-1").
